cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaProjectBuilder.java

```diff
@@ -147,9 +148,37 @@ public NvaProject build() {
                    .withHealthProjectData(extractHealthProjectData(cristinProject))
                    .withApprovals(extractApprovals(cristinProject.getApprovals()))
                    .withExemptFromPublicDisclosure(cristinProject.getExemptFromPublicDisclosure())
+                   .withCreator(extractCreator(cristinProject.getCreator()))
                    .build();
     }
 
+    private NvaContributor extractCreator(CristinPerson creator) {
+        return Optional.ofNullable(creator)
+            .stream().peek(this::addRoleDataIfMissing)
+            .flatMap(NvaProjectBuilder::generateRoleBasedContribution)
+                   .findAny()
+                   .or(() -> creatorWithoutAffiliation(creator))
+                   .orElse(null);
+    }
+
+    private Optional<NvaContributor> creatorWithoutAffiliation(CristinPerson creator) {
+        return Optional.ofNullable(creator)
+                   .filter(presentCreator -> nonNull(presentCreator.getCristinPersonId()))
+                   .map(presentCreator -> {
+                       var creatorWithoutAffiliation = new NvaContributor();
+                       creatorWithoutAffiliation.setIdentity(Person.fromCristinPerson(presentCreator));
+                       return creatorWithoutAffiliation;
+                   });
+    }
+
+    private void addRoleDataIfMissing(CristinPerson cristinPerson) {
+        if (isNull(cristinPerson.getRoles())) {
+            cristinPerson.setRoles(emptyList());
+        } else {
+            cristinPerson.getRoles().forEach(role -> role.setRoleCode(PRO_CREATOR));
```

Comment:
we do not get role code from upstream. To match the NvaContributor's existing logic without errors, we need a role code. Then i can reuse existing models and method without creating new ones only for creator